public/docs/_examples/lifecycle-hooks/dart/lib/do_check_component.dart

```diff
@@ -0,0 +1,107 @@
+// #docregion
+import 'dart:convert';
+
+import 'package:angular2/core.dart';
+
+class Hero {
+  String name;
+  Hero(this.name);
+  Map<String, dynamic> toJson() => {'name': name};
+}
+
+@Component(
+  selector: 'do-check',
+  template: '''
+  <div class="hero">
+    <p>{{hero.name}} can {{power}}</p>
+
+    <h4>-- Change Log --</h4>
+    <div *ngFor="let chg of changeLog">{{chg}}</div>
+  </div>
+  ''',
+  styles: const [
+    '.hero {background: LightYellow; padding: 8px; margin-top: 8px}',
+    'p {background: Yellow; padding: 8px; margin-top: 8px}'
+  ])
+class DoCheckComponent implements DoCheck, OnChanges {
+  @Input() Hero hero;
+  @Input() String power;
+
+  bool changeDetected = false;
+  List<String> changeLog = [];
+
+  String oldHeroName = '';
+  String oldPower = '';
+  int oldLogLength = 0;
+  int noChangeCount = 0;
+
+  // #docregion ng-do-check
+  ngDoCheck() {
```

Comment:
It has an implicit return type of `dynamic`, thus matching the signature of the interface method it is overriding. I would suggest leaving it like this.